### PR TITLE
Issue #35: MMD tools module

### LINKED ISSUE #35: Make the code extracted from py_mmd_tools a proper submodule

### PR DESCRIPTION
**Pull request by vkbo**
**Summary**:

This PR removes the xml_utils from py_mmd_tools and implements the needed code directly in the Worker class. The schema validator object is also created in the App class and just passed to the Worker class. This means that the 15 second init time is now only an issue when starting, and not on each API request.

The tests have been updated to allow for this change, and a mock xsd schema file created to make the App tests run faster.

The check_mmd file still remains, but is now in an mmd_tools folder.

**Related issue**:

Closes #35

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: dmci/api/app.py [hunk 1]
```diff
@@ -40,6 +41,13 @@ def __init__(self):
             logger.error("Parameter distributor_cache in config is not set")
             sys.exit(1)
 
+        if self._conf.mmd_xsd_path is None:
+            logger.error("Parameter mmd_xsd_path in config is not set")
+            sys.exit(1)
+
+        # Create the XML Validator Object
+        self._xsd_obj = etree.XMLSchema(etree.parse(self._conf.mmd_xsd_path))
```
Comment (mortenwh) [inline dmci/api/app.py:49]: so this takes 15 seconds..?

**Review by mortenwh — APPROVED** [review]
Looks good!

**Review by johtoblan — APPROVED** [review]
:+1:

File: dmci/api/app.py (hunk 1 above)
Comment (vkbo) [reply dmci/api/app.py:49]: Yes

**Pull request merged**